src/main/java/com/dev/cinema/dao/UserDaoImpl.java

```diff
@@ -21,6 +22,7 @@ public UserDaoImpl(SessionFactory sessionFactory) {
             CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
             CriteriaQuery<User> criteriaQuery = criteriaBuilder.createQuery(User.class);
             Root<User> root = criteriaQuery.from(User.class);
+            root.fetch("roles", JoinType.LEFT);
```

Comment:
why do we use JoinType.LEFT?

Reply:
well, I agree that the situation when we have user without role doesn't seem possible, but as I can recall we need to specify JoinType.LEFT to be sure we've got rid of LazyInitializationException. But it may be that I just misunderstood something and it was necessary to specify JoinType whatever it is ('cos we can use root.fetch("roles") without specifying JoinType)

Reply:
done